akka-persistence/src/main/scala/akka/persistence/serialization/PayloadSerializer.scala

```diff
@@ -94,18 +103,22 @@ final class PayloadSerializer(val system: ExtendedActorSystem)
     val protoCompositeMetadata = mf.CompositeMetadata.parseFrom(bytes)
 
     val metadataEntries =
-      protoCompositeMetadata.getPayloadsList.iterator.asScala.map { persistentPayload =>
+      protoCompositeMetadata.getPayloadsList.iterator.asScala.flatMap { persistentPayload =>
         val manifest =
           if (persistentPayload.hasPayloadManifest)
             persistentPayload.getPayloadManifest.toStringUtf8
           else ""
 
+        // if the metadata type is unknown to the deserializing service, we do not want to fail deserialization
+        // of the event as a whole, so such metadata payloads are ignored
         serialization
           .deserialize(persistentPayload.getPayload.toByteArray, persistentPayload.getSerializerId, manifest)
-          .get
+          .toOption
```

Comment:
Shall we have a marker trait to define that the metadata is mandatory and should fail deserialization instead of skipping? Then we have it as default to ignore, but can mark certain things, like `ReplicatedEventMetadata`.

Reply:
We only have a serializer id and a manifest, no class on the deserializing side when unknown metadata type so we couldn't know that the marker is applied?

Reply:
yeah, didn't think (that far)